scipy/sparse/_dok.py

```diff
@@ -101,27 +103,80 @@ def keys(self):
     def values(self):
         return self._dict.values()
 
-    def get(self, key, default=0.):
-        """This overrides the dict.get method, providing type checking
-        but otherwise equivalent functionality.
-        """
+    def get(self, key, default=0.0):
+        """This provides dict.get method functionality with type checking"""
+        if key in self._dict:
+            return self._dict.get(key, default)
+        if isintlike(key) and self.ndim == 1:
+            key = (key,)
```

Comment:
I'm not sure we want to do this, because the 1d DOK format stores its keys as plain integers.

We might need a fully separate codepath for the 1d case in this function.

Reply:
I forgot to unwrap the key in the 1d case before the final line.

Also, this now allows negative indices for DOK in `get`. That brings it closer to `A[key]`.

I can remove the handling of negative indices -- and can make a separate path for 1d if you'd prefer.